tests/integration/docker_utils/conftest.py

```diff
@@ -17,10 +17,15 @@ def _check_skip(client: ContainerClient):
 
 
 @pytest.fixture(
-    params=[CmdDockerClient(), SdkDockerClient()], ids=["CmdDockerClient", "SdkDockerClient"]
+    params=[CmdDockerClient, SdkDockerClient], ids=["CmdDockerClient", "SdkDockerClient"]
 )
-def docker_client(request):
-    client = request.param
+def docker_client_class(request):
+    yield request.param
+
+
+@pytest.fixture
+def docker_client(docker_client_class):
+    client = docker_client_class()
```

Comment:
With the cache, creating a new docker client will yield in a bit of a memory leak, since the lru cache will hold a reference to "self", which leads to the client not removed. Before, we only had one client for all tests, now we have a new one for every test, and the CmdDockerClients will not be garbage collected. Again, not a huge leak, so probably not worth following up, but still. We could scope the docker client to class level, which would reduce that issue.

Reply:
Good catch! I added the scope :)